portal/src/hook/useAppConfigForm.ts

```diff
@@ -45,6 +48,7 @@ export function useAppConfigForm<State>(
   const [isUpdating, setIsUpdating] = useState(false);
   const [updateError, setUpdateError] = useState<unknown>(null);
   const [isSubmitted, setIsSubmitted] = useState(false);
+  const [canSave, setCanSave] = useState<boolean | undefined>(initialCanSave);
```

Comment:
Can we receive `canSave` from argument directly? I mean something like

```
interface Options {
  validate?: (state: State) => APIError | null;
  canSave?: boolean;
}

function useAppConfigForm(..., options?: Options) {
  const canSave = options?.canSave;
  const validate = options?.validate;
}
```

Then we do not need to expose `setCanSave`, and do not need to use `useEffect` to sync `canSave` in the hook caller.

Reply:
Sure, let me revise :)

Reply:
Oh, I found we cannot update this way. In most of the case, the canSave state is determined by the current form state, and the form state is created by  `useAppConfigForm`. So it is difficult for us to provide canSave in `useAppConfigForm` argument. Seems having `setCanSave` is more flexible. Thought?

Reply:
OK. let's keep what we have now.